### ISSUE TITLE
SQLite CREATE TRIGGER... WHEN expression is unparsable without parentheses

### ISSUE THREAD

**Issue by kgutwin** [sqlfluff/sqlfluff#6100]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff reports "unparsable section" when providing a `WHEN expression` to a SQLite `CREATE TRIGGER` expression. If the expression is put in parentheses, however, parsing is successful. The parentheses should be optional, not required.

Example SQL:

```sql
CREATE TRIGGER x
AFTER INSERT ON y
WHEN new.z IS NULL   -- putting this expression in parens allows parsing
BEGIN
UPDATE y SET z = TRUE WHERE rowid = new.rowid;
END;
```

### Expected Behaviour

```
All Finished 📜 🎉! 
```

### Observed Behaviour

```
== [stdin] FAIL                                                                                                                                                                    
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TRIGGER x\nAFTER INSERT ON y\nWHEN ...'
WARNING: Parsing errors found and dialect is set to 'sqlite'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```
echo "CREATE TRIGGER x
AFTER INSERT ON y
WHEN new.z IS NULL  
BEGIN
UPDATE y SET z = TRUE WHERE rowid = new.rowid;
END;" | sqlfluff lint --dialect sqlite -
```

### Dialect

sqlite

### Version

sqlfluff, version 3.1.0
Python 3.11.9

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
